javascript/src/main/components/Schedule/PersonalScheduleTable.js

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import BootstrapTable from 'react-bootstrap-table-next';
+import { Button } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
+
+const PersonalScheduleTable = ({ data, deleteSchedule }) => {
+
+  const history = useHistory();
+  const renderEditButton = (id) => {
+    return (
+      <Button data-testid={`edit-button-${id}`} onClick={() => { 
+        console.log("Updating:", id);
+        history.push(`/schedule/update/${id}`) ;
+      }}>Edit</Button>
+    )
+  }
+

```

Comment:
I wonder why we need an entire file that is an exact duplicate of `ScheduleTable` except for the name, and the line:

```
 formatter: (cell, row) => <a href={"/schedule/" + row.id}> {cell} </a>
```

Especially when having that  be a link to is entirely harmless.

I would suggest deleting this file, and replacing references to `PersonalScheduleTable` with a reference to `ScheduleTable`.  (I recall suggesting to someone on the team that they just reuse `ScheduleTable` and pass in an array with an single object in it.)

Reply:
The reason for this was because when I didn't have a specific version of ScheduleTable, the link on the show page would send the user to /schedule/schedule/(row.id), which would be an invalid page.

Reply:
Actually never mind, I think I forgot to put a backslash in my code, which would explain a lot